README.md

```diff
@@ -8,19 +8,69 @@ This should use the [Go Modules Tool](https://blog.golang.org/using-go-modules)
 
 ## Setup
 
+### Environment Variables
+
 You will need to copy the `.env.template` file to `.env` and fill in the appropriate values.
 See the server `/var/www/html/go/credentials.json` for the credential values.
 
 Once you have done that, when you run the script, you will be prompted to authorize the app and create the appropriate tokens. (Or you can download `token.json` from the server.
 
 All this is to keep credentials out of version control.
 
+### Database
+
+Once the environment variables have been set (particularly `BEB_ENV`), 
+you will need to initialize the database with a privileged database user (probably `root`):
+```shell
+setup/setup.sql.sh | sudo mysql -uroot -p
+```
+
+The `BEB_ENV` specifies a postfix for the database name depending on the environment (development, test, production, etc).
+
+### Database Tools
+
+We use [Gorm](https://gorm.io/docs/) for ORM and other database queries.
+
+### Database Migrations
+
+Schema Migrations are held in the `migrations` folder. You will need to run them once upon every update.
+
+```shell
+go run migrations/*.go
+```
+
+You can create a new migration with:
+```shell
+go run setup/setup.go create_migration [Name your migration]
+```
+
+This action will create a new file in the `migrations` folder. It will have an object with two actions `up` and `down`.
+The `up` action will be where you will put [migration code](https://gorm.io/docs/migration.html).
+The `down` action undoes your migration, which is useful for testing the migration code and undoing changes from a bad deploy.
+
 ## Run Script
 
 `go run *.go`
 
 You need to run all the local `main` package files together with the `run` command.
 
+## Testing
+
+Run current tests:
+
+```
+go test ./controllers/
+```
+
+The controller tests stub the model layer of the code using [mockgen](https://github.com/golang/mock), rather than using database seeds (at the moment).
+
+To create or update the stubs when one has updated the model layer, run (inserting the correct model file):
+```
+mockgen -source=models/notifications.go -destination=mock_models/notifications.g
```

Comment:
Should there have been an "o" at the end of his line for `-destination=mock_models/notifications.go`?  I guess it really does not matter as this is only an example of how to create or update a mock model.

Reply:
Fixed